docs/src/examples/components/Input/Types/InputInvertedExample.shorthand.tsx

```diff
@@ -11,14 +11,19 @@ const InputExample = () => (
     >
       <Input inverted placeholder="Inverted color input..." />
     </Grid>
-    <Grid
-      styles={({ theme: { siteVariables } }) => ({
-        backgroundColor: siteVariables.colorScheme.default.background,
-        padding: '20px',
-      })}
-    >
-      <Input placeholder="Default input..." />
-    </Grid>
+    {/* TODO: figure this out - changed because of restricting props styles to be only object :\ */}
```

Comment:
Why do we require that the `styles` prop be only an object?

Reply:
Because if we use `mergeStyles` it is always the same function, hence resulting in wrong hash key (although the parameters are different). This is not an issue when we are passing the styles from parent => child, as by the time they get to the factories, they are already resolved. It's literally only for the styles prop... I don't have other idea of how to solve this at this moment..

Reply:
Hm, OK.  Let's revisit.  This one seems like a complex rabbit hole.